### DISCUSSION TITLE
ILI9341 conecction to T3 V1.6.1 LILYGO

### DISCUSSION THREAD

**Discussion by PA2J** [dl9rdz/rdz_ttgo_sonde#255] [Q&A]
Hi,

cannot get it to work,  my ili9341 display will not show something  
in the SondeServer Hardware configuration display type selection = 3 and pinout from TTGO
MISO=19
MOSI=23
SCK=2
CS=15
RS=13
LED=25
Reset = not connected
as settings are to be done for SDA, guess is ment for ILI9225 display i wonder if i did mis out on something like a user.h foor the ILI9341 display in Arduino IDE??

any idea or correction is welcome

73
Jasper
PA2J

**Comment by dl9rdz** [comment]
Not sure what you exactly configured in the config. For your pinout, you need to manually configure your pins.

MISO is not needed, can be left not connected
MOSI is SDA, so "OLED SDA/TFT SDA" in the config needs to be set to 23 with your wiring
SCK, CS and RS also needs to be configured that way
Reset should be connected, and corresponding pin configured in "OLED RST/TFT RST"
Probably, LED should simply be connected to 3V3. (If using GPIO25, you could try setting "power control pin" to value 153 (==128+25), but not sure if the io pins can deliver sufficient power)

You should also decrease the SPI speed in the hardware config. 10 Mhz is within the datasheet spec, 20 Mhz is outside spec but works fine for me in practice, the default value 40 Mhz is fine for ILI9225, but will not work at all for most ILI9341.

**Reply by PA2J** [reply]
The way i got it to work in Hardware Configuration
MOSI=21
CLK=22
RST=15
RS=12
CS=13
TTF SPI set to 5Mhz  > will not work (this was the trigger i needed)
Display type = 3 and LED connected to 3.3V pin on the TTGO

Thanks putting me on the right track Hansi
73

Jasper PA2J

**Reply by wrevenaugh** [reply]
What is "RS" - I don't see an "RS" on either the Board or the ILI9341.  Thanks!

**Comment by dl9rdz** [comment]
"RS" is "register select" and an ILI9341 with 4-pin SPI interface should have it...

It is actually "data select/register select" and on some boards might be labelled "DC" or "DC/RS"

In the data sheet it actually is "Data/Command selection input (D/CX)"

**Reply by wrevenaugh** [reply]
Thanks @dl9rdz , that's correct, it's called "DC" on my display.   Might be better off calling it "DC" in the UI?  Also, as Jasper pointed out above, the key was getting the Freq set to 5,000,000 for me.  It was all garbled with anything higher. Thanks again, and thanks @PA2J !

  BTW @dl9rdz , any progress on the Imperial measurement stuff?  We'd sure love that here in the States! (hint hint) ;-)    I'm not anywhere near as good a coder as you are but I sure would like to help out if I can, or if you can tell me where to look in your OVERWHELMING code (lol), I'd be happy to give it a shot!  Bill

**Reply by lenalotti** [reply]
On my TFT it is also label DC witch is the same as RS on SPI ports, unfortunately they are not always labeled the same. And yes, I am using a "standard" ESP32 board, for the LORA I connected an Ra-02 module (sx1278 chip), and for localistion a NEO-6M GPS-Module.

**Reply by wrevenaugh** [reply]
@lenalotti  ha ha ha ha DUH!  (I'm as dumb as a box of rocks, lol) Here I am connecting an external  display and I can't even imagine that someone has actually hooked up an external Lora chip to an ESP32 Dev Module (not to mention I'm surrounded by a handful of them connected up to all kinds of sensors and screens!) Very cool! so then another Question: Now that I have my gorgeous HUGE display connected I want to use these alternative screens that @Bazjo has made but I cant figure out how to do it.  I'm using the "screens editor" and in screens.txt I see "### Alternative display layouts based on https://gist.github.com/bazjo"  but i am just getting the old standard black and white simple screens.  I've even updated to the latest version "devel0514" but still nothing.  Have you done this, or do you know where to look to figure out how to get these?  Thanks! Bill

**Reply by dl9rdz** [reply]
For a large display, you should first select in "Screen config ...." either 4 or 5 (for landscape or portrait mode of the display). You then (probably) need to reboot to make the board load screens4.txt

Then, you can, again in the config, change the line "Display screens", for example to "5,6,7" for the @bazjo´s screens

**Reply by wrevenaugh** [reply]
@dl9rdz Thank you so much! ITS AWESOME!
